packages/valory/skills/check_stop_trading_abci/behaviours.py

```diff
@@ -0,0 +1,162 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the behaviours for the check stop trading skill."""
+
+import math
+from typing import Any, Generator, Set, Type, cast
+
+from packages.valory.contracts.mech.contract import Mech as MechContract
+from packages.valory.skills.abstract_round_abci.base import get_name
+from packages.valory.skills.abstract_round_abci.behaviour_utils import BaseBehaviour
+from packages.valory.skills.abstract_round_abci.behaviours import AbstractRoundBehaviour
+from packages.valory.skills.check_stop_trading_abci.models import CheckStopTradingParams
+from packages.valory.skills.check_stop_trading_abci.payloads import CheckStopTradingPayload
+from packages.valory.skills.check_stop_trading_abci.rounds import (
+    CheckStopTradingRound,
+    CheckStopTradingAbciApp,
+)
+from packages.valory.skills.staking_abci.behaviours import (
+    StakingInteractBaseBehaviour,
+    WaitableConditionType,
+)
+from packages.valory.contracts.service_staking_token.contract import StakingState
+
+
+class CheckStopTradingBehaviour(StakingInteractBaseBehaviour):
+    """A behaviour that checks stop trading conditions."""
+
+    matching_round = CheckStopTradingRound
+
+    @property
+    def mech_request_count(self) -> int:
+        """Get the liveness period."""
+        return self._mech_request_count
+
+    @mech_request_count.setter
+    def mech_request_count(self, mech_request_count: int) -> None:
+        """Set the liveness period."""
+        self._mech_request_count = mech_request_count
+
+    def _get_mech_request_count(self) -> WaitableConditionType:
+        """Get the mech request count."""
+        status = yield from self.contract_interact(
+            contract_address=self.params.mech_contract_address,
+            contract_public_id=MechContract.contract_id,
+            contract_callable="get_requests_count",
+            data_key="requests_count",
+            placeholder=get_name(CheckStopTradingBehaviour.mech_request_count),
+            address=self.synchronized_data.safe_contract_address,
+        )
+        return status
+
+    @property
+    def is_first_period(self) -> bool:
+        """Return whether it is the first period of the service."""
+        return self.synchronized_data.period_count == 0
+
+    @property
+    def params(self) -> CheckStopTradingParams:
+        """Return the params."""
+        return cast(CheckStopTradingParams, self.context.params)
+
+    def is_staking_kpi_met(self) -> WaitableConditionType:
```

Comment:
Why does this return `WaitableConditionType` since you do not wait for `True` from the caller method?

Reply:
This call is mimmicking similar calls from staking_abci:

https://github.com/valory-xyz/trader/blob/69af494465a802fe80f4dd5ce6d53f0b1f14e010/packages/valory/skills/staking_abci/behaviours.py#L247

https://github.com/valory-xyz/trader/blob/69af494465a802fe80f4dd5ce6d53f0b1f14e010/packages/valory/skills/staking_abci/behaviours.py#L255

Reply:
Yes, but these are consumed by the `wait_for_condition_with_sleep` method.